x-pack/plugins/ml/public/ui_actions/open_in_single_metric_viewer_action.tsx

```diff
@@ -0,0 +1,97 @@
+import { i18n } from '@kbn/i18n';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import type { EmbeddableApiContext } from '@kbn/presentation-publishing';
+import { apiIsOfType } from '@kbn/presentation-publishing';
+import type { UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import { ML_APP_LOCATOR, ML_PAGES } from '../../common/constants/locator';
+import type { MlEmbeddableBaseApi, SingleMetricViewerEmbeddableApi } from '../embeddables';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '../embeddables';
+
+import type { MlCoreSetup } from '../plugin';
+
+export interface OpenInSingleMetricViewerActionContext extends EmbeddableApiContext {
+  embeddable: SingleMetricViewerEmbeddableApi;
+}
+
+export const OPEN_IN_SINGLE_METRIC_VIEWER_ACTION = 'openInSingleMetricViewerAction';
+
+export function isSingleMetricViewerEmbeddableContext(
+  arg: unknown
+): arg is OpenInSingleMetricViewerActionContext {
+  return (
+    isPopulatedObject(arg, ['embeddable']) &&
+    apiIsOfType(arg.embeddable, ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE)
+  );
+}
+
+const getTimeRange = (embeddable: MlEmbeddableBaseApi): TimeRange | undefined => {
+  return embeddable.timeRange$?.getValue() ?? embeddable.parentApi?.timeRange$?.getValue();
+};
+
+export function createOpenInSingleMetricViewerAction(
+  getStartServices: MlCoreSetup['getStartServices']
+): UiActionsActionDefinition<OpenInSingleMetricViewerActionContext> {
+  return {
+    id: 'open-in-single-metric-viewer',
+    type: OPEN_IN_SINGLE_METRIC_VIEWER_ACTION,
+    getIconType(): string {
+      return 'visTable';
+    },
+    getDisplayName() {
+      return i18n.translate('xpack.ml.actions.openInSingleMetricViewerTitle', {
+        defaultMessage: 'Open in Single Metric Viewer',
+      });
+    },
+    async getHref(context): Promise<string | undefined> {
+      const [, pluginsStart] = await getStartServices();
+      const locator = pluginsStart.share.url.locators.get(ML_APP_LOCATOR)!;
+
+      if (isSingleMetricViewerEmbeddableContext(context)) {
+        const { embeddable } = context;
+        const {
+          input: { jobIds, query, selectedEntities },
+        } = embeddable;
+
+        return locator.getUrl(
+          {
+            page: ML_PAGES.SINGLE_METRIC_VIEWER,
+            // @ts-ignore entities is not compatible with SerializableRecord
+            pageState: {
+              timeRange: getTimeRange(embeddable),
+              refreshInterval: {
+                display: 'Off',
+                pause: true,
+                value: 0,
+              },
+              jobIds,
+              query,
+              entities: selectedEntities,
+            },
+          },
+          { absolute: true }
+        );
+      }
+    },
+    async execute(context) {
+      if (!context.embeddable) {
+        throw new Error('Not possible to execute an action without the embeddable context');
+      }
```

Comment:
```suggestion
      if (!isSingleMetricViewerEmbeddableContext(context)) {
        throw new IncompatibleActionError();
      }
```

Reply:
Updated in bc6fb2ca9b1833fffc6f524af3055ef317a18b47